.github/workflows/ci.yaml

```diff
@@ -0,0 +1,176 @@
+# This workflow will install Python dependencies, run tests and lint with a variety of Python versions
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: CI
+
+on:
+  push:
+    branches: [ $default-branch, add_github_actions ]
+  pull_request:
+    branches: [ $default-branch ]
+
+env:
+  DEFAULT_PYTHON: 3.7
+
+jobs:
+  lint-flake8:
+    name: Check flake8
+    runs-on: ubuntu-latest
+    steps:
+    - name: Check out code from Github
+      uses: actions/checkout@v2
+
+    - name: Set up Python ${{ env.DEFAULT_PYTHON }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ env.DEFAULT_PYTHON }}
+
+    - name: Cache dependencies
+      uses: actions/cache@v2
+      with:
+        path: ~/.cache/pip
+        key: ${{ runner.os }}-${{ env.DEFAULT_PYTHON }}-pip-flake8
```

Comment:
This may not re-cache the newer version of `flake8` installed by `pip`.

Reply:
After reviewing how the caching action works, the current method is flawed for all jobs. As of now even the `requirements_dev.txt` cache will _not_ update if there's an upstream version bump. We'll need to pin the versions in order for the current method to works so that the hash of the requirements file changes during bumps.

As for the separate `flake8` and `coverage` jobs It's probably better to either:

1. Don't bother caching during these runs.
2. Go further and cache a fully built venv instead of just the pip packages, and then use that venv in all the jobs.

Reply:
Went with option 2 above, creating and caching a venv for each python version.